### DISCUSSION TITLE
Undefined namespace in PHP

### DISCUSSION THREAD

**Discussion by unglud** [JetBrains/Qodana#91] [Q&A]
Hi!

I have a lot of `Undefined namespace` errors for libs that come from the composer. Doc said the composer should run in Qodana. It's hard to judge, but it seems like for every file. Is there a way to debug and see if something is wrong with the composer run? An example here is `Undefined namespace 'Aws'` line 6. File `src/Api/Aws/AwsFactory.php`
```
<?php
declare(strict_types=1);

namespace App\Api\Aws;

use Aws\Sdk;

/**
 * Class AwsFactory
 *
 * @package App\Api\Aws
 */
class AwsFactory {
    public static function build($configOverride = []): Sdk {
        $sharedConfig = [
            "region" => "eu-central-1",
            "version" => "latest",
            "retries" => ["mode" => "adaptive", "max_attempts" => 3]
        ];

        return new Sdk(array_merge($sharedConfig, $configOverride));
    }
}
```
Here is composer
```
{
    "version": "1.1",
    "require": {
        "aws/aws-sdk-php": "~3.173"
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "config": {
        "process-timeout": 0,
        "sort-packages": true,
        "autoloader-suffix": "fulfin",
        "notify-on-install": false,
        "allow-plugins": {
            "dg/composer-cleaner": true
        }
    }
}
```

**Comment by andrii-pez** [comment]
For me there was a problem in Jenkins job related to php version which was used during composer installation. Somehow the latest version of Qodana used incorrect php version(8.0 instead of 8.1). So that packages were not installed and Qodana showed many errors for `Undefined namespace` and classes
